From ea1a4f6134acf027884434988e265f3fa4930ddd Mon Sep 17 00:00:00 2001
From: make shi <b15407@freescale.com>
Date: Thu, 8 Nov 2012 15:23:33 +0800
Subject: [PATCH 2247/2463] ENGR00233051-02 Mx6 USB: msl headfile for OTG
 modulization

MSL headfile part change

-Add and remove some function define in usb.h

Signed-off-by: make shi <b15407@freescale.com>
---
 arch/arm/mach-mx6/usb.h |   15 ++++++++++++---
 1 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/usb.h b/arch/arm/mach-mx6/usb.h
index f796e7d..b57104b 100644
--- a/arch/arm/mach-mx6/usb.h
+++ b/arch/arm/mach-mx6/usb.h
@@ -27,8 +27,6 @@ extern int fsl_usb_host_init(struct platform_device *pdev);
 extern void fsl_usb_host_uninit(struct fsl_usb2_platform_data *pdata);
 extern int gpio_usbotg_utmi_active(void);
 extern void gpio_usbotg_utmi_inactive(void);
-
-extern void __init mx6_usb_dr_init(void);
 extern bool usb_icbug_swfix_need(void);
 extern void __init mx6_usb_h2_init(void);
 extern void __init mx6_usb_h3_init(void);
@@ -38,7 +36,8 @@ extern void mx6_set_host3_vbus_func(driver_vbus_func);
 extern void mx6_set_host2_vbus_func(driver_vbus_func);
 extern void mx6_set_host1_vbus_func(driver_vbus_func);
 extern void mx6_set_otghost_vbus_func(driver_vbus_func);
-extern void mx6_set_usb_host1_vbus_func(driver_vbus_func *driver_vbus);
+extern void mx6_get_otghost_vbus_func(driver_vbus_func *driver_vbus);
+extern void mx6_get_host1_vbus_func(driver_vbus_func *driver_vbus);
 extern struct platform_device anatop_thermal_device;
 extern struct platform_device mxc_usbdr_otg_device;
 extern struct platform_device mxc_usbdr_udc_device;
@@ -73,3 +72,13 @@ extern void __iomem *imx_otg_base;
 		.iobase = soc ## _USB_ ## hs ## _BASE_ADDR, 	\
 		.irq = soc ## _INT_USB_ ## hs,				\
       }
+#define imx_fsl_usb2_otg_data_entry_single(soc)				\
+		  {					\
+			  .iobase = soc ## _USB_OTG_BASE_ADDR,  \
+			  .irq = soc ## _INT_USB_OTG,			  \
+		  }
+#define imx_fsl_usb2_udc_data_entry_single(soc) \
+		  {				\
+			  .iobase = soc ## _USB_OTG_BASE_ADDR, \
+			  .irq = soc ## _INT_USB_OTG,			  \
+		  }
-- 
1.7.7.4

